daemon/heartbeat.c

```diff
@@ -1097,9 +1147,28 @@ receive_hb()
                 invalid_packet_recvd = TRUE;
                 time_invalid_packet_recvd = now;
 
+                char ip_address[INET6_ADDRSTRLEN];
+                if (!inet_ntop(from.sa.sa_family, &from.sa, ip_address, sizeof ip_address))
+                {
+                    sprintf(ip_address, "<invalid_ip>");
```

Comment:
It would be better to use `snprintf` to avoid getting this flagged in the future. This relies on `<invalid_ip>` being smaller than `INET6_ADDRSTRLEN`, which is true but can't be proven by just looking at this code.

Reply:
Mmm... Otherwise can simplify like this:

```C
                if (!inet_ntop(from.sa.sa_family, &from.sa, ip_address, sizeof ip_address))
                    *ip_address = '\0';

...

             log_message(MTC_LOG_WARNING,
                    "HB: invalid packet received from (%s:%d).\n",
                    *ip_address ? ip_address : "<invalid_ip>", port);
```

What do you think? It removes completely a copy.

Reply:
I thought about 

```
char ip_address[INET6_ADDRSTRLEN] = "<invalid_ip>";
```
but I am not sure that this can be later overwritten as desired.

Reply:
I hesitated to do that, I was not sure that the buffer was not modified in case of error, and that the default value remained unchanged. But it's ok, we can use this solution. :wink: I will repush.